janus_core/processing/observables.py

```diff
@@ -1,72 +1,292 @@
-        *args : tuple
-            Additional positional arguments passed to getter.
-        **kwargs : dict
-            Additional kwargs passed getter.
 
         Returns
         -------
-        float
-            The stress component in GPa units.
+        list[float]
+            The stress components in GPa units.
         """
+        sliced_atoms = atoms[self.atoms_slice]
+        sliced_atoms.calc = atoms.calc
         return (
-            atoms.get_stress(include_ideal_gas=self.include_ideal_gas, voigt=True)[
-                self._index
-            ]
+            sliced_atoms.get_stress(
+                include_ideal_gas=self.include_ideal_gas, voigt=True
+            )
             / units.GPa
-        )
+        )[self._indices]
+
+
+StressHydrostatic = Stress(components=["xx", "yy", "zz"])
+StressShear = Stress(components=["xy", "yz", "zx"])
+
+
+# pylint: disable=too-few-public-methods
+class Velocity(Observable, ComponentMixin):
+    """
+    Observable for per atom velocity components.
+
+    Parameters
+    ----------
+    components : list[str]
+        Symbols for velocity components, x, y, z.
+    atoms_slice : list[int] | SliceLike | None = None
+        List or slice of atoms to observe velocities from.
+    """
+
+    def __init__(
+        self,
+        *,
+        components: list[str],
+        atoms_slice: list[int] | SliceLike | None = None,
+    ):
+        """
+        Initialise the observable from a symbolic str component and atom index.
+
+        Parameters
+        ----------
+        components : list[str]
+            Symbols for tensor components, x, y, and z.
+        atoms_slice : Union[list[int], SliceLike]
+            List or slice of atoms to observe velocities from.
+        """
+        ComponentMixin.__init__(self, components={"x": 0, "y": 1, "z": 2})
+        self._set_components(components)
+
+        Observable.__init__(self, len(components))
+
+        self.atoms_slice = atoms_slice if atoms_slice else slice(0, None, 1)
+
+    def __call__(self, atoms: Atoms) -> list[float]:
+        """
+        Get the velocity components for correlated atoms.
+
+        Parameters
+        ----------
+        atoms : Atoms
+            Atoms object to extract values from.
+
+        Returns
+        -------
+        list[float]
+            The velocity values.
+        """
+        return atoms.get_velocities()[self.atoms_slice, :][:, self._indices].flatten()
+
+    def value_count(self, n_atoms: int | None = None) -> int:
+        """
+        Count of values returned by __call__.
+
+        Parameters
+        ----------
+        n_atoms : int | None
+            Atom count to expand atoms_slice.
+
+        Returns
+        -------
+        int
+            The number of values returned by __call__.
+        """
+        if isinstance(self.atoms_slice, list):
+            return len(self.atoms_slice) * self.dimension
+        return slicelike_len_for(self.atoms_slice, self.n_atoms) * self.dimension
```

Comment:
It might be best to move the list case into `slicelike_len_for` so we can have a clean interface here. Maybe `selector_len` or something in that case?

Reply:
done, but this also made me realise slicelike_len_for is not defined due to the import guarding (ultimately a circular import issue from before). @ElliottKasoar I have moved ```CorrelationKwargs``` into ```correlator.py``` so there is no longer an issue with ```Observable``` and ```SliceLike``` circular imports.